src/client/utils/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react'
+
+function getStorageValue(key: string, defaultValue: string | boolean) {
+  // getting stored value
+  const saved = localStorage.getItem(key)
+  const initial = saved && JSON.parse(saved)
+  return initial || defaultValue
+}
+
+export const useLocalStorage = (
+  key: string,
+  defaultValue: string | boolean,
+) => {
+  const [value, setValue] = useState(() => {
+    return getStorageValue(key, defaultValue)
+  })
+
+  useEffect(() => {
+    // storing input name
+    localStorage.setItem(key, JSON.stringify(value))
+  }, [key, value])
```

Comment:
not necessary to use an effect there, no?

We can just wrap the `setValue` function and call `localStorage.setItem(key, JSON.stringify(value))` inside of it.